src/sugar3/graphics/progressicon.py

```diff
@@ -14,86 +14,171 @@
 from sugar3.graphics import style
-
-
-class ProgressIcon(Gtk.DrawingArea):
-    """Display the progress filling the icon.
+ 
+class ProgressIcon(Gtk.DrawingArea): '''UI interface for Progress Icon.'''
+ '''Display the progress filling the icon.
 
     This class is compatible with the sugar3.graphics.icon.Icon class.
 
     Call update(progress) with the new progress to update the icon.
-
     The direction defaults to 'vertical', in which case the icon is
     filled from bottom to top.  If direction is set to 'horizontal',
     it will be filled from right to left or from left to right,
     depending on the system's language RTL setting.
 
-    """
-    def __init__(self, icon_name, pixel_size, stroke_color, fill_color,
+ '''
+    def __init__(self, icon_name, pixel_size, stroke_color, fill_color, 
                  direction='vertical'):
-        Gtk.DrawingArea.__init__(self)
-
-        self._icon_name = icon_name
-        self._direction = direction
-        self._progress = 0
+        ''' 
+        Initializes the ProgressIcon class
+        
+        Parameters:
+           icon_name - The name of icon progressicon uses.
+           pixel_size - It explains basic geometry of an image. 
+           stroke_color - Stroke color means border color. Here, it is for progressicon.
+           fill_color - The inside color of progressicon. 
+        
+        Example :
+          
+           class Box:
+           def area(self):
+           return self.width * self.height
+
+           def __init__(self, width, height):
+           self.width = width
+           self.height = height
+
+           # Create an instance of Box.
+           x = Box(10, 2)
+
+           # Print area.
+           print(x.area())
+        
+        '''
+        
+        Gtk.DrawingArea.__init__(self) '''here, init is instructor, self represents instance of __init__'''
+
+        self._icon_name = icon_name 
+        self._direction = direction 
+        self._progress = 0 '''Initial Value of progress is zero'''
 
         self._stroke = get_surface(
             icon_name=icon_name, width=pixel_size, height=pixel_size,
             stroke_color=stroke_color,
             fill_color=style.COLOR_TRANSPARENT.get_svg())
-
+            '''
+            Sets stroke color (border color of progressicon) to transparent. 
+            '''
+       
         self._fill = get_surface(
             icon_name=icon_name, width=pixel_size, height=pixel_size,
             stroke_color=style.COLOR_TRANSPARENT.get_svg(),
             fill_color=fill_color)
-
-        self.connect("draw", self.__draw_cb)
+            '''
+            Sets fill color to transparent. 
+            '''
+        self.connect("draw", self.__draw_cb) '''Connect ---> Draw to self.__draw_cb'''
 
     def __draw_cb(self, widget, cr):
-        allocation = widget.get_allocation()
-
+     '''
+     Example:
+         def __draw_cb(self, da, cr): 
+         cr.set_source_rgb(1, 1, 1)
+         cr.paint()
+     '''
```

Comment:
As Quozl had stated before, this function is intended to be private (i.e. not to be accessed outside this class). The example for the invoking of this function seems unnecessary.